include/picongpu/plugins/output/images/Visualisation.hpp

```diff
@@ -157,286 +159,654 @@ struct typicalFields < 5 >
-            if (image(imageCell).x() > float_X(1.0)) image(imageCell).x() = float_X(1.0);
-            if (image(imageCell).y() < float_X(0.0)) image(imageCell).y() = float_X(0.0);
-            if (image(imageCell).y() > float_X(1.0)) image(imageCell).y() = float_X(1.0);
-            if (image(imageCell).z() < float_X(0.0)) image(imageCell).z() = float_X(0.0);
-            if (image(imageCell).z() > float_X(1.0)) image(imageCell).z() = float_X(1.0);
-        }
+        );
     }
 };
 
 namespace vis_kernels
 {
 
+/** divide each cell by a value
+ *
+ * @tparam T_numWorkers number of workers
+ * @tparam T_blockSize number of elements which will be handled
+ *                     within a kernel block
+ */
+template<
+    uint32_t T_numWorkers,
+    uint32_t T_blockSize
+>
 struct DivideAnyCell
 {
+    /** derive particle values
+     *
+     * @tparam T_Mem pmacc::DataBox, type of the on dimensional memory
+     * @tparam T_Type divisor type
+     * @tparam T_Acc alpaka accelerator type
+     *
+     * @param acc alpaka accelerator
+     * @param mem memory[in,out] to manipulate, must provide the `operator[](int)`
+     * @param n number of elements in mem
+     * @param divisor divisor for the division
+     */
     template<
-        typename Mem,
-        typename Type,
+        typename T_Mem,
+        typename T_Type,
         typename T_Acc
     >
     DINLINE void operator()(
         T_Acc const & acc,
-        Mem mem,
+        T_Mem mem,
         uint32_t n,
-        Type divisor
+        T_Type divisor
     ) const
     {
-        uint32_t tid = blockIdx.x * blockDim.x + threadIdx.x;
-        if (tid >= n) return;
+        using namespace mappings::threads;
+
+        constexpr uint32_t numWorkers = T_numWorkers;
 
-        const float3_X FLT3_MIN = float3_X(FLT_MIN, FLT_MIN, FLT_MIN);
-        mem[tid] /= (divisor + FLT3_MIN);
+        uint32_t const workerIdx = threadIdx.x;
+
+        using SupercellDomCfg = IdxConfig<
+            T_blockSize,
+            numWorkers
+        >;
+        // each virtual worker works on a cell
+        ForEachIdx< SupercellDomCfg > forEachCell( workerIdx );
+
+        forEachCell(
+            [&](
+                uint32_t const linearIdx,
+                uint32_t const
+            )
+            {
+                uint32_t tid = blockIdx.x * T_blockSize + linearIdx;
+                if( tid >= n )
+                    return;
+
+                float3_X const FLT3_MIN = float3_X::create( FLT_MIN );
+                mem[ tid ] /= ( divisor + FLT3_MIN );
+            }
+        );
     }
 };
 
 
+/** convert channel value to RGB color
+ *
+ * @tparam T_numWorkers number of workers
+ * @tparam T_blockSize number of elements which will be handled
+ *                     within a kernel block
+ */
+template<
+    uint32_t T_numWorkers,
+    uint32_t T_blockSize
+>
 struct ChannelsToRGB
 {
+    /** convert each element to a RGB color
```

Comment:
OPTIONAL:
[grammatically correct](http://www.grammar.com/a-vs-an-when-to-use/) it would be: a**n** RGB